azkaban-hadoop-security-plugin/src/main/java/azkaban/security/Credential.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright 2017 LinkedIn Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ *
+ */
+
+package azkaban.security;
+
+/**
+ * Facilitate to register user secret key into custom credential object.
+ */
+public interface Credential {
```

Comment:
I'd probably call this CredentialProvider or something like that

Reply:
`CredentialProvider ` Sounds better. Thanks.